### PR TITLE
[CX-2074] Add scheduler and worker

### PR DESCRIPTION
**Pull request by cfernhout**
*Add a label `automerge` to automatically merge the PR after approval and passed checks.*

### CODE REVIEW

File: charts/datafold/templates/_server.tpl [hunk 1]
```diff
@@ -1,14 +1,3 @@
...
-    Return the server host
-*/}}
-{{- define "datafold.server.server" -}}
-{{- if not .Values.global.server.server -}}
-{{ printf "%s-%s" .Release.Name "redis" }}
-{{- else -}}
-{{ .Values.global.server.server }}
-{{- end -}}
-{{- end -}}
-
```
Comment (cfernhout) [inline charts/datafold/templates/_server.tpl:11]: Unused

File: charts/datafold/charts/scheduler/templates/_helpers.tpl [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+Selector labels
+*/}}
+{{- define "scheduler.selectorLabels" -}}
+app.kubernetes.io/name: {{ include "scheduler.name" . }}
+{{- end }}
+
+{{/*
+Create the name of the service account to use
+*/}}
+{{- define "scheduler.serviceAccountName" -}}
```
Comment (gtoonstra) [inline charts/datafold/charts/scheduler/templates/_helpers.tpl:56]: Scheduler doesnt make k8s calls (API), so we don't need a service account I guess.

File: charts/datafold/charts/scheduler/templates/_helpers.tpl [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+Common labels
+*/}}
+{{- define "scheduler.labels" -}}
+helm.sh/chart: {{ include "scheduler.chart" . }}
+app.kubernetes.io/component: web-app
+{{ include "scheduler.selectorLabels" . }}
+{{- if .Chart.AppVersion }}
+app.kubernetes.io/version: {{ .Chart.AppVersion | quote }}
+{{- end }}
+{{ include "datafold.labels" . }}
```
Comment (gtoonstra) [inline charts/datafold/charts/scheduler/templates/_helpers.tpl:43]: Nice. Took that along immediately.

File: charts/datafold/charts/scheduler/templates/hpa.yaml [hunk 4]
```diff
@@ -0,0 +1,32 @@
+{{- if .Values.autoscaling.enabled }}
+apiVersion: autoscaling/v2
```
Comment (gtoonstra) [inline charts/datafold/charts/scheduler/templates/hpa.yaml:2]: autoscaling. Let's remove for scheduler.

File: charts/datafold/charts/scheduler/templates/service.yaml [hunk 5]
```diff
@@ -0,0 +1,14 @@
+apiVersion: v1
+kind: Service
```
Comment (gtoonstra) [inline charts/datafold/charts/scheduler/templates/service.yaml:2]: Scheduler shouldn't have a port. remove?

File: charts/datafold/charts/scheduler/values.yaml [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+image:
+  pullPolicy: IfNotPresent
+  tag: ""
+
+imagePullSecrets:
+  - name: datafold-docker-secret
+nameOverride: ""
+fullnameOverride: ""
+
+serviceAccount:
```
Comment (gtoonstra) [inline charts/datafold/charts/scheduler/values.yaml:12]: remove?

File: charts/datafold/charts/scheduler/values.yaml [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+  name: ""
+
+podAnnotations: {}
+podLabels: {}
+
+podSecurityContext: {}
+
+securityContext: {}
+
+service:
```
Comment (gtoonstra) [inline charts/datafold/charts/scheduler/values.yaml:25]: remove?

File: charts/datafold/charts/scheduler/values.yaml [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+  enabled: true
+  className: ""
+  annotations: {}
+  path: /
+  pathType: ImplementationSpecific
+  tls: []
+
+resources: {}
+
+autoscaling:
```
Comment (gtoonstra) [inline charts/datafold/charts/scheduler/values.yaml:38]: remove?

File: charts/datafold/charts/worker/templates/hpa.yaml [hunk 9]
```diff
@@ -0,0 +1,32 @@
+{{- if .Values.autoscaling.enabled }}
```
Comment (gtoonstra) [inline charts/datafold/charts/worker/templates/hpa.yaml:1]: Yes, let's keep for a while.

File: charts/datafold/templates/_server.tpl [hunk 10]
```diff
@@ -1,14 +1,3 @@
-{{/*
-    Return the server host
-*/}}
-{{- define "datafold.server.server" -}}
-{{- if not .Values.global.server.server -}}
-{{ printf "%s-%s" .Release.Name "redis" }}
```
Comment (gtoonstra) [inline charts/datafold/templates/_server.tpl:6]: is aangepast in andere PR ergens, want fout.

File: charts/datafold/values.yaml [hunk 11]
```diff
@@ -21,6 +23,8 @@ global:
   server:
     port: 5000
     hostName: k8s.dev.local
+  scheduler:
```
Comment (gtoonstra) [inline charts/datafold/values.yaml:26]: Does scheduler have a port?

File: charts/datafold/values.yaml [hunk 12]
```diff
@@ -36,6 +40,15 @@ dfshell:
 
 server:
   install: true
+  command: "server"
```
Comment (gtoonstra) [inline charts/datafold/values.yaml:43]: I kept these local to keep the config close to where it's used. If it's not more globally used elsewhere and not used at this level, the preference I chose was to keep it as close as possible to the chart using it.

File: jeeves/dev.py [hunk 13]
```diff
@@ -8,6 +8,7 @@ class EnvVar(enum.StrEnum):
     DATAFOLD_K8S_SECRETFILE = "DATAFOLD_K8S_SECRETFILE"
     DATAFOLD_DEPLOY_NAME = "DATAFOLD_DEPLOY_NAME"
     DATAFOLD_DEV_OVERRIDES = "DATAFOLD_DEV_OVERRIDES"
+    DATAFOLD_DB_HOSTPATH = "DATAFOLD_DB_HOSTPATH"
```
Comment (gtoonstra) [inline jeeves/dev.py:11]: Ik had deze juist uit de dev gehaald, omdat deze niet gebruikt wordt verder. In ee nandere PR wordt deze juist eruit gehaald.  Op die manier lekken we geen dev details naar een prod deployment.

File: jeeves/dev.py [hunk 14]
```diff
@@ -36,6 +37,8 @@ def _common_args():
         '-f',
         os.getenv(EnvVar.DATAFOLD_DEV_OVERRIDES),
         '--set',
+        f'global.hostPath={os.getenv(EnvVar.DATAFOLD_DB_HOSTPATH)}',
```
Comment (gtoonstra) [inline jeeves/dev.py:40]: verwijderen?

File: secrets/secrets.yaml [hunk 15]
```diff
@@ -14,11 +14,28 @@ secrets:
     user: "postgres"
     password: "postgres"
     database: "postgres"
+  redis:
+    password: "redis-pw"
+
+server:
```
Comment (gtoonstra) [inline secrets/secrets.yaml:20]: secrets is only used for linting, so we can keep these empty probably.

**Review by gtoonstra — CHANGES_REQUESTED** [review]

**Pull request merged**